### PR TITLE
:sparkles: (llm|lld): delete local app data on uninstall and uninstallAll

### PR DESCRIPTION
**Pull request by valpinkman**
### ✅ Checklist

- [x] `npx changeset` was attached.
- [x] **Covered by automatic tests.** 
- [x] **Impact of the changes:** 
  - delete app data when restore is finished (wip: handle UserRefusedOnDevice case)
  - updated the apprunner state to know when to delete app data backup during uninstall
  - refactor execWithTransport in its own file
  - update lld to use new flow
  - update llm to use new flow

### 📝 Description

This PR makes sure to remove local app data when uninstalling app through the manager on LLD/LLM + all changes required in LLC to make it work

### ❓ Context

- **JIRA or GitHub link**: [LIVE-13454] [LIVE-12688] 

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-13454]: https://ledgerhq.atlassian.net/browse/LIVE-13454?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/ledger-live-common/src/device/use-cases/appDataBackup/restoreAppData.ts [hunk 1]
```diff
@@ -51,14 +64,24 @@ export function restoreAppData(
...
 
-          // NOTE: DELETE DATA
-
           subscriber.next({
             type: RestoreAppDataEventType.AppDataRestored,
           });
-          subscriber.complete();
+        }),
+        // Delete the app data from the storage
+        switchMap(() => deleteAppData(appName, deviceModelId, storageProvider)),
```
Comment (jiyuzhuang) [inline libs/ledger-live-common/src/device/use-cases/appDataBackup/restoreAppData.ts:72]: Interaction with data storage in this file is counter to the initial pattern.
This `restoreAppData` as well as `backupAppData` are dedicated to data stream programming. I think that the data deletion use case should be enchained in `restoreAppDataUseCase` where all the parameters are already present.

Comment (jiyuzhuang) [reply libs/ledger-live-common/src/device/use-cases/appDataBackup/restoreAppData.ts:72]: In addition, the deleting the local data is not part of the data restoration workflow. It is an operation independent and should not be injected into normal restoration workflow.

Comment (valpinkman) [reply libs/ledger-live-common/src/device/use-cases/appDataBackup/restoreAppData.ts:72]: make sense, i will fix this

File: libs/ledger-live-common/src/device/use-cases/execWithTransport.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import Transport from "@ledgerhq/hw-transport";
+import { Exec } from "../../apps";
+import installApp from "../../hw/installApp";
+import installAppWithRestore from "../../hw/installAppWithRestore";
+import uninstallApp from "../../hw/uninstallApp";
+import uninstallAppWithBackup from "../../hw/uninstallAppWithBackup";
+
+export const execWithTransport =
```
Comment (ofreyssinet-ledger) [inline libs/ledger-live-common/src/device/use-cases/execWithTransport.ts:8]: [COULD] Maybe it's worth adding a bit of TSDoc to explain what this does, and even potentially rename it to something less mysterious...

**Review by ofreyssinet-ledger — CHANGES_REQUESTED** [review]

**Review by ofreyssinet-ledger — CHANGES_REQUESTED** [review]
Some things to fix in async unit tests.
Also I think the name of the `deleteAppDataBackup` is maybe a bit too vague.
And I'm not sure I understand where the actual deletion of locally stored app data is done for app uninstallation, `deleteAppDataUseCaseDI` is never called in uninstall flows as far as I understand 🤔

**Comment by valpinkman** [comment]
> Some things to fix in async unit tests. Also I think the name of the `deleteAppDataBackup` is maybe a bit too vague. And I'm not sure I understand where the actual deletion of locally stored app data is done for app uninstallation, `deleteAppDataUseCaseDI` is never called in uninstall flows as far as I understand 🤔

Things are done here https://github.com/LedgerHQ/ledger-live/pull/7744/files#diff-cadd4d39653d7214faf8f5dd226d1a6e9c4fc35c45e66a08787170bcc9097d71 to know how to handle each case. (the state)
And there is no "deletion" when uninstalling https://github.com/LedgerHQ/ledger-live/pull/7744/files#diff-64aaf9798ce046311b2798bcaa0ebd94cf9c3308faf867ac000a861a6c150a87R19 we just skip the backup.

But fair point I will rename `deleteAppDataBackup` to make more sense

File: libs/ledger-live-common/src/device/use-cases/appDataBackup/restoreAppData.test.ts [hunk 3]
```diff
@@ -29,65 +29,85 @@ describe("restoreAppData", () => {
...
+        });
+        expect(events[3]).toEqual({
+          type: RestoreAppDataEventType.Progress,
+          data: expect.any(Number),
+        });
+        expect(events[4]).toEqual({
+          type: RestoreAppDataEventType.AppDataRestored,
+        });
+        done();
+      },
```
[the remark this answers is not in the document]
Comment (valpinkman) [reply libs/ledger-live-common/src/device/use-cases/appDataBackup/restoreAppData.test.ts:66]: done

File: libs/ledger-live-common/src/device/use-cases/appDataBackup/restoreAppData.test.ts [hunk 4]
```diff
@@ -29,65 +29,85 @@ describe("restoreAppData", () => {
...
+          type: RestoreAppDataEventType.Progress,
+          data: expect.any(Number),
+        });
+        expect(events[3]).toEqual({
+          type: RestoreAppDataEventType.Progress,
+          data: expect.any(Number),
+        });
+        expect(events[4]).toEqual({
+          type: RestoreAppDataEventType.AppDataRestored,
+        });
```
[the remark this answers is not in the document]
Comment (valpinkman) [reply libs/ledger-live-common/src/device/use-cases/appDataBackup/restoreAppData.test.ts:105]: done

**Review by ofreyssinet-ledger — APPROVED** [review]
nice 👍

**Pull request merged**